### PR TITLE
Add ValidatingAdmissionPolicyBinding resource

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
Ref: https://issues.redhat.com/browse/ACM-14837

[](https://issues.redhat.com/browse/ACM-14837)

### Related Issue
https://issues.redhat.com/browse/ACM-14837

### Description of changes
- Added ValidatingAdmissionPolicyBinding resource

### CODE REVIEW

File: pkg/transforms/vapbinding.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+type VapBindingResource struct {
+	node Node
+}
+
+func VapBindingResourceBuilder(v *unstructured.Unstructured) *VapBindingResource {
```
Comment (yiraeChristineKim) [inline pkg/transforms/vapbinding.go:21]: I could use `k8s.io/api` v0.30 but I had to change prow, other libraries(a lot) and go version. Therefore, I rather chose to use `unstruntured`

**Comment by yiraeChristineKim** [comment]
@mprahl @dhaiducek @JustinKuli @l

**Comment by yiraeChristineKim** [comment]
@jlpadilla

File: pkg/transforms/vapbinding.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	node.Properties["validationActions"], _, _ = unstructured.NestedStringSlice(v.Object, "spec", "validationActions")
+	node.Properties["policyName"], _, _ = unstructured.NestedString(v.Object, "spec", "policyName")
+
+	owners := v.GetOwnerReferences()
+	fromGK := false
+
+	for _, o := range owners {
+		if strings.HasPrefix(o.APIVersion, "constraints.gatekeeper.sh") {
+			fromGK = true
```
Comment (jlpadilla) [inline pkg/transforms/vapbinding.go:42]:
```suggestion
			fromGK = true
			break
```

File: pkg/transforms/vapbinding.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		}
+	}
+
+	node.Properties["fromGatekeeper"] = fromGK
```
Comment (jlpadilla) [inline pkg/transforms/vapbinding.go:46]: Do we expect users to search by this property? If no, then change it to `_fromGatekeeper`.

Comment (jlpadilla) [reply pkg/transforms/vapbinding.go:46]: Would `_ownedByGatekeeper` be more descriptive?

File: pkg/transforms/vapbinding.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/*
+IBM Confidential
+OCO Source Materials
+(C) Copyright IBM Corporation 2019 All Rights Reserved
+The source code for this program is not published or otherwise divested of its trade secrets,
+irrespective of what has been deposited with the U.S. Copyright Office.
+Copyright (c) 2020 Red Hat, Inc.
+*/
+// Copyright Contributors to the Open Cluster Management project
```
Comment (jlpadilla) [inline pkg/transforms/vapbinding.go:9]: Only the single line copyright is needed for new files.
```suggestion
// Copyright Contributors to the Open Cluster Management project
```

File: pkg/transforms/vapbinding.go (hunk 3 above)
Comment (yiraeChristineKim) [reply pkg/transforms/vapbinding.go:46]: great

**Review by mprahl — APPROVED** [review]

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold
@yiraeChristineKim you can unhold if this is ready to merge. Thanks for the contribution!

**Pull request merged**